pkg/workflow/workflow.go

```diff
@@ -255,6 +265,16 @@ func newEngine(ctx monitorContext.Context, wfCtx wfContext.Context, w *workflow,
 		rk:            w.rk,
 		stepStatus:    stepStatus,
 		stepDependsOn: stepDependsOn,
+		stepTimeout:   make(map[string]time.Time),
+	}
+}
+
+func setStepStatus(statusMap map[string]common.StepStatus, status common.StepStatus) {
```

Comment:
```suggestion
func copyStepStatus(statusMap map[string]common.StepStatus, status common.StepStatus) {
```

Reply:
set means you're changing something, copy is better here that you're just copy and override

Reply:
I actually set the status to the map but not copy so I think it's `set` here.